enhancements/cluster-logging/elasticsearch-percentage-index-management.md

```diff
@@ -0,0 +1,156 @@
+
+## Motivation
+
+Currently our managed Elasticsearch cluster is limited to only specify rollover and deletion policies based on the age of indices which does not translate easily into disk utilization which can mean unforseen expenses for users trying to fit their usage into a specific time window of supporting index age. If instead they can tell our index management to start deleting the oldest indices once usage reaches a certain percentage this gives a better user experience.
+
+### Goals
+The specific goals of this proposal are:
+
+* Outline the addition to the API to provide a means to configure usage for our policies
+
+* Update the current cronjobs to be able to make decisions based on usage
+
+We will be successful when:
+
+* Users are able to also configure their policies with "usage" instead of just age.
+
+* We also contribute this feature to opendistro so that we stay aligned with what they offer (for future adoptions).
+
+### Non-Goals
+
+* For our ES 6.x cluster we will not be implementing this using Opendistro due to gaps in that version of the plugin
+
+## Proposal
+
+Currently we offer a mechanism for users to configure their index lifecycles based on time only. This can be difficult for customer when trying to anticipate their disk usage, especially if they are charged for the storage they use and need to scale up because they were unable to anticipate a large amount of logs being processed.
+
+This seeks to offer a mechanism by which one can denote the usage percentage (either of each node or of the total cluster) for indicating which indices should be deleted over.
+
+### User Stories
+
+#### As an user of EO, I want to be able to configure my delete policies based on a percent usage
+
+Currently we only offer a mechanism to delete indicies based on the age of the index.
+
+### Implementation Details
+
+#### Assumptions
+
+* This will be going out with our ES6 operator which means it will be a part of our ES 6.x cluster
+
+* It will be possible to configure phases based on both usage and time. We will trigger on whichever comes first.
+
+#### Security
+
+No additional security concerns need to be addressed by this.
+
+### Risks and Mitigations
+
+## Design Details
+
+We will add a field that is used by the delete phase. We will delete on either MinAge or Usage, whichever is evaluated as true first.
+
+### index_management_types.go
+
+We will add a usage field for our phases to use:
+
+```
+// +k8s:openapi-gen=true
+type IndexManagementDeletePhaseSpec struct {
+	// The minimum age of an index before it should be deleted (e.g. 10d)
+	//
+	MinAge TimeUnit `json:"minAge, optional"`
+
+  // The percent disk usage to wait for before deleting indices for this alias (e.g. 75)
```

Comment:
Which is more convenient here - percentage or absolute size? I'm not sure of the answer.
Could support both by treating "75" or "75%" as a percent, and "75G" as a size.

Reply:
it seems that the request was for a percentage and I think that would actually make it more simple for users. If we wanted to support a static size I would suggest we have a field that is specific for that -- like a k8s resource request

Reply:
Why not make this more explicit and telling what it represents `MaxUsagePercent` or so that it is in agreement with age should it be `MinUsagePercent`?

Reply:
is "PercentUsage" be explicit and clear enough?

Reply:
I don't feel theres a need to say "Min" or "Max"... we have a way for the user to tell us this is how much percent usage they want to allow. So more than that and we curate, less than that and we don't trigger.